machines/vic/reification/appliance.go

```diff
@@ -0,0 +1,416 @@
+				Device: &types.VirtualIDEController{
+					VirtualController: types.VirtualController{
+						VirtualDevice: types.VirtualDevice{
+							Key: 200,
+						},
+					},
+				},
+			},
+		},
+	}
+	spec.DeviceChange = append(spec.DeviceChange, netDevices...)
+	return spec, nil
+}
+
+func (d *Dispatcher) findAppliance(conf *config.Config) (*vm.VirtualMachine, error) {
+	if ovm, err := d.session.Finder.VirtualMachine(d.ctx, conf.DisplayName); err != nil {
+		_, ok := err.(*find.NotFoundError)
+		if !ok {
+			err = errors.Errorf("Failed to query appliance (%s): %s", conf.DisplayName, err)
+			return nil, err
+		} else {
+			log.Debugf("Appliance is not found")
+			return nil, nil
+		}
+	} else {
+		newVM := vm.NewVirtualMachine(d.ctx, d.session, ovm.Reference())
+		// workaround here. We lost the value set in ovm cause we wrap the object to another type
+		newVM.InventoryPath = ovm.InventoryPath
+		return newVM, nil
+	}
+}
+
+func (d *Dispatcher) createAppliance(conf *config.Config) error {
+	log.Infof("Creating appliance on target")
+
+	spec, err := d.createApplianceSpec(conf)
+	if err != nil {
+		log.Errorf("Unable to create appliance spec: %s", err)
+		return err
+	}
+
+	// create test VM
+	info, err := tasks.WaitForResult(d.ctx, func(ctx context.Context) (tasks.ResultWaiter, error) {
+		return d.session.Folders(ctx).VmFolder.CreateVM(ctx, *spec, d.vchPool.ResourcePool, d.session.Host)
+	})
+
+	if err != nil {
+		log.Errorf("Unable to create appliance VM: %s", err)
+		return err
+	}
+	if info.Error != nil || info.State != types.TaskInfoStateSuccess {
+		log.Errorf("Create appliance reported: %s", info.Error.LocalizedMessage)
+	}
+
+	// get VM
+	vm, err := d.findAppliance(conf)
+	if err != nil || vm == nil {
+		err = errors.Errorf("Failed query back appliance: %s", err)
+		return err
+	}
+
+	// update the displayname to the actual folder name used
+	if d.vmPathName, err = vm.FolderName(d.ctx); err != nil {
+		log.Errorf("Failed to get canonical name for appliance: %s", err)
+		return err
+	}
+	log.Debugf("vm folder name: %s", d.vmPathName)
+	log.Debugf("vm inventory path: %s", vm.InventoryPath)
+	var unitNumber int32
+	unitNumber = -1
+
+	// set component execution parameters into guestinfo
+	spec.DeviceChange = []types.BaseVirtualDeviceConfigSpec{
+		// currently we're hardcoded to boot from a base ISO so add that
+		&types.VirtualDeviceConfigSpec{
+			Operation: types.VirtualDeviceConfigSpecOperationAdd,
+			Device: &types.VirtualCdrom{
+				VirtualDevice: types.VirtualDevice{
+					Key:           -2,
+					ControllerKey: 200,
+					UnitNumber:    &unitNumber,
+					Backing: &types.VirtualCdromIsoBackingInfo{
+						VirtualDeviceFileBackingInfo: types.VirtualDeviceFileBackingInfo{
+							FileName: fmt.Sprintf("[%s] %s/appliance.iso", conf.ImageDatastoreName, d.vmPathName),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	files := "/var/tmp/images/ /var/log/vic/"
+
+	var vicadmintlsargs string
```

Comment:
Agree to avoid unnecessary vm.reconfigure. 
But here, I think if we upload iso files before VM is created, the files might locate in different directory with the new VM, cause in some scenarios, e.g. vSAN, the VM folder name is different with vm name.
And based on that, there is one parameter vm.InventoryPath is configured after VM is created.
I understand vSAN is not current priority, but I think the installation order should be same for different environments.